src/ui-test/java_completion_support_test.ts

```diff
@@ -0,0 +1,173 @@
+		after(_clean(CAMEL_CONTEXT_JAVA));
+
+		it('Duplicate endpoint options are filtered out', async function () {
+			const editor = new TextEditor();
+
+			await editor.typeTextAt(9, URI_POSITION, 'timer');
+			contentAssist = await waitUntilContentAssistContains(contentAssist, editor, 'timer:timerName');
+			const timer = await contentAssist.getItem('timer:timerName');
+			await timer.click();
+
+			await editor.typeTextAt(9, URI_POSITION + 15, '?');
+			contentAssist = await waitUntilContentAssistContains(contentAssist, editor, 'delay');
+			const delay = await contentAssist.getItem('delay');
+			await delay.click();
+
+			await editor.typeTextAt(9, URI_POSITION + 26, '&de');
+			contentAssist = await editor.toggleContentAssist(true) as ContentAssist;
+			await new WaitUntil().assistHasItems(contentAssist, DefaultWait.TimePeriod.DEFAULT);
+			const filtered = await contentAssist.hasItem('delay');
+
+			assert.isFalse(filtered);
+			await editor.toggleContentAssist(false);
+		});
+	});
+
+	describe('Diagnostics for Camel URIs', function () {
+
+		const EXPECTED_ERROR_MESSAGE = 'Invalid duration value: 1000r';
+
+		beforeEach(_setup(CAMEL_CONTEXT_JAVA));
+		afterEach(_clean(CAMEL_CONTEXT_JAVA));
+
+		it('LSP diagnostics support for XML DSL', async function () {
+			this.retries(3);
+			const editor = new TextEditor();
+
+			await editor.typeTextAt(9, URI_POSITION, 'timer');
+			contentAssist = await waitUntilContentAssistContains(contentAssist, editor, 'timer:timerName');
+			const timer = await contentAssist.getItem('timer:timerName');
+			await timer.click();
+
+			await editor.typeTextAt(9, URI_POSITION + 15, '?');
+			contentAssist = await waitUntilContentAssistContains(contentAssist, editor, 'delay');
+			const delay = await contentAssist.getItem('delay');
+			await delay.click();
+
+			await editor.typeTextAt(9, URI_POSITION + 26, 'r');
+			const problemsView = await utils.openView('Problems');
+
+			await driver.wait(async function () {
+				const innerMarkers = await problemsView.getAllMarkers(MarkerType.Error);
+				return innerMarkers.length > 0;
+			}, DefaultWait.TimePeriod.MEDIUM);
+			const markers = await problemsView.getAllMarkers(MarkerType.Error);
+			assert.isNotEmpty(markers, 'Problems view does not contains expected error');
+
+			const errorMessage = await markers[0].getText();
+			assert.include(errorMessage, EXPECTED_ERROR_MESSAGE);
+			await new BottomBarPanel().toggle(false); // close Problems View
+		});
+	});
+
+	async function waitUntilContentAssistContains(contentAssist: ContentAssist, editor: TextEditor, expectedContentAssist: string) {
+		await DefaultWait.sleep(2000);
+		await driver.wait(async function () {
+			contentAssist = await editor.toggleContentAssist(true) as ContentAssist;
+			const hasItem = await contentAssist.hasItem(expectedContentAssist);
+			if (!hasItem) {
+				await editor.toggleContentAssist(false);
+			}
+			return hasItem;
+		}, DefaultWait.TimePeriod.DEFAULT);
+		return contentAssist;
+	}
```

Comment:
This would be nice to extract thi smethod to an util test class to factorize with https://github.com/camel-tooling/camel-lsp-client-vscode/blob/a6ab899a3c040225d85239233fed6bf198846a3d/src/ui-test/tests/xml_completion_support.test.ts#L195

Reply:
+1
but that is something I would like to see in another PrR or as part of next UI test for Yaml DSL support